### PR TITLE
feat(authentication): allow authentication via login/logout (dev) and PKI (prod)

### PR DESCRIPTION
**Pull request by irtiza-rizvi**
AMLOS-550

**Description**     
As a user, I want to be able to log into the application

**Acceptance Criteria**   
Allow user to login with Login and Password
Allow user to logout
Allow user to login with PKI Certificate
 
**How to test code**    
Pull `authentication_module` from `ozp-backend`
Pull `authentication_module` from `frontend`

_Dev_
Navigate to the home page.  Ensure that you are redirect to login page.
Log in as any user.  Ensure that you are redirected to the home page, the storefront loads properly, and the user's name displays in the nav bar.  Ensure (via Firebug / Dev Tools in chrome) that the JWT token is part of session storage and persisted as a cookie.
Log out via hamburger menu in nav bar.  Ensure that session storage and cookies are cleared.

_Prod_
Navigate to the home page.  Ensure that you can access the application normally without requiring a login.
Ensure that logout button is not available.

### CODE REVIEW

**Comment by flesher1** [comment]
This breaks ```npm run dev -- --env.API_URL="http://ip:8001"```

The APIs use a config file. I suggest we standardize which method we use.

**Comment by flesher1** [comment]
This breaks ```npm run dev -- --env.API_URL="http://ip:8001"```

The APIs use a config file. I suggest we standardize which method we use.

:+1:  Styles look good, everything works as expected. APIs function as expected.

File: src/services/auth.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import $ from 'jquery';
+import { API_URL } from '../config';
```
Comment (blynch11p) [inline src/services/auth.js]: As @flesher1 mentioned, use 
`var { API_URL } = require('Config');`

**Comment by blynch11p** [comment]
I made the fix to use the different Config import, so token_auth looks for the API_URL defined and not localhost.

:+1:

**Comment by irtiza-rizvi** [comment]
Thanks @blynch11p.  I'll make sure to do that moving forward.

**Pull request merged**